### PR TITLE
Add folders for Lighting Presets & Improve Admin Interface

### PR DESCRIPTION
**Pull request by cherry-john**
## Adds a nested folder structure to lighting presets

image

image

## Moves the web dashboard to a tiled structure 

image

## Doesn't require instant reboots for all changes on web dash

image

# TODO

- [x] CSS Styling tweaks @Jbithell ( :heart: )
- [x] Work out why builds are failing (https://github.com/Jbithell/ParadisePi/pull/11)

### CODE REVIEW

File: admin/main.js [hunk 1]
```diff
@@ -325,19 +384,26 @@ $(document).on('submit', 'form.preset-form[data-table]', function () {
 });
 
 //add new preset
-$('#lxNew').click(function (){
+$('#lxNew').on('click', function(){
```
Comment (Jbithell) [inline admin/main.js:387]: Noice

File: main.js [hunk 2]
```diff
@@ -147,6 +155,7 @@ async function initDatabases() {
       table.string('universe');
       table.json('setArguments');
       table.integer("fadeTime").defaultTo(null);
+      table.integer('folderId').defaultTo(1);
```
Comment (Jbithell) [inline main.js:158]: Ah so for legacy databases this is a breaking change?

File: admin/main.js [hunk 3]
```diff
@@ -10,6 +10,13 @@ jQuery.fn.addChild = function(html = "<div class='oscDiv'>")
 //sets which mixer type to get info for
 let mixer = "x32";
 
+//store current list of folders
+let folders = [];
+
+//Store universe data
+let firstUniverse = 1;
```
Comment (Jbithell) [inline admin/main.js:17]: Is this needed @cherry-john  - can't see where it's used

Comment (cherry-john) [reply admin/main.js:17]: Used to set bounds on old LX cards, Is legacy code that I moved to the top

File: admin/main.js (hunk 1 above)
Comment (cherry-john) [reply admin/main.js:387]: apparently .click is depreciated or something

File: main.js (hunk 2 above)
Comment (cherry-john) [reply main.js:158]: Yes, but that's the case with most of the updates to this project and just something we accepted :)

File: admin/main.js (hunk 3 above)
Comment (Jbithell) [reply admin/main.js:17]: Ahh awesome

**Comment by Jbithell** [comment]
@cherry-john hopefully this builds now (he says)

**Comment by Jbithell** [comment]
image

image

# TODO

- [ ] Fix styling of channel check functions

**Comment by Jbithell** [comment]
@cherry-john this is big enough already - let's merge and I'll open a new PR to move the channel check styling into the folder styling

**Review by Jbithell — APPROVED** [review]

**Pull request merged**